crates/uniswapx-rs/src/order.rs

```diff
@@ -285,6 +400,131 @@ fn resolve_decay(
     }
 }
 
+impl NonlinearDutchDecay {
+
+    /// Calculates the decayed amount based on the current block and the defined curve
+    pub fn decay(
+        &self,
+        start_amount: Uint<256, 4>,
```

Comment:
maybe cleaner to define type alias for Uint<256, 4>?

Reply:
Good idea